clients/src/test/java/org/apache/kafka/common/memory/WeakMemoryPoolTest.java

```diff
@@ -45,6 +45,8 @@ public void testNullRelease() {
 
     @Test
     public void testAllocationMemorySize() {
+        // Start from clean state to avoid flakiness
+        System.gc();
```

Comment:
?!

Reply:
When tests are run in parallel, it can so happen that this test case might fail thereby introducing flakiness. With this change, I'm hoping to reduce the flakiness.